packages/resolve-url-loader/lib/engine/postcss.js

```diff
@@ -28,37 +28,12 @@ function process(sourceFile, sourceContent, params) {
     sourceContent.replace(ORPHAN_CR_REGEX, ' $1') :
     sourceContent;
 
-  // prepend file protocol to all sources to avoid problems with source map
-  return postcss([
-    postcss.plugin('postcss-resolve-url', postcssPlugin)
-  ])
-    .process(correctedContent, {
-      from: fileProtocol.prepend(sourceFile),
-      map : params.outputSourceMap && {
-        prev          : !!params.absSourceMap && fileProtocol.prepend(params.absSourceMap),
-        inline        : false,
-        annotation    : false,
-        sourcesContent: true  // #98 sourcesContent missing from output map
-      }
-    })
-    .then(result => ({
-      content: result.css,
-      map    : params.outputSourceMap ? fileProtocol.remove(result.map.toJSON()) : null
-    }));
-
   /**
    * Plugin for postcss that follows SASS transpilation.
    */
-  function postcssPlugin() {
-    return function applyPlugin(styles) {
-      styles.walkDecls(eachDeclaration);
-    };
-
-    /**
-     * Process a declaration from the syntax tree.
-     * @param declaration
-     */
-    function eachDeclaration(declaration) {
+  const postcssPlugin = {
+    postcssPlugin: 'postcss-resolve-url',
+    Declaration: (declaration) => {
```

Comment:
This code has been locking up E2E tests. The test eventually crashes Node with "out of memory" error.

Finally worked out from [this migration guide](https://evilmartians.com/chronicles/postcss-8-plugin-migration) that...
> plugins will re-visit all changed or added nodes. You should check if your transformations were already applied and if that is the case—ignore those nodes.

So here the code needs to somehow keep track of visited declarations.
```
() => {
  postcssPlugin: 'postcss-resolve-url',
  prepare: () => {
    const visited = new Set();
    return {
      Declaration: (declaration) => {
        ...
      }
    };
  }
};
```

I will fix this and make some other tweaks post-merge.